src/services/loggingexport/loggingexportService.js

```diff
@@ -0,0 +1,172 @@
+	 *				"logData": "[\"SERVICE LIFECYCLE: STATE CHANGE: Service initializing\",\"windowService\"]",
+	 *													//JSON encoded erray of message and data components of the log message
+	 *													//N.B. maybe be prefixed by string "*** Logging Error: ""
+	 *				"logTimestamp": 1544090028391.6226	//Log message timestamp for ordering use
+	 *			}, 
+	 *			{...},
+	 *			...
+	 *		]
+	 * @private
+	 */
+	this.addToBatch = function (dataArr) {
+		if (dataArr) {
+			for (let m = 0; m < dataArr.length; m++) {
+				if (CAPTURE_LOG_CATEGORIES[dataArr[m].category] && CAPTURE_LOG_LEVELS[dataArr[m].logType]) {
+					//TODO: add any necessary message format changes here
+					logBatch[currBatchSize++] = dataArr[m];
+				} 
+				// else {
+				// 	console.log("discarding message",  dataArr[m]);
+				// }
+			}
+
+			if(currBatchSize >= BATCH_SIZE){
+				self.transmitBatch();
+			} else if (!timeout) { //always transmit batch within minimum timeout
+				timeout = setTimeout(self.transmitBatch, TIMEOUT_MILLISECS);
+			}
+		} else {
+			Logger.warn("Tried to add an invalid data array to log batch",  dataArr);
+		}
+	}
+
+	/**
+	 * Transmit the batch to the remote log collection endpoint.
+	 * @private
+	 */
+	this.transmitBatch = function () {
+		clearTimeout(timeout);
+		//trim batch array to length
+		let toTransmit = logBatch;
+		toTransmit.splice(currBatchSize, logBatch.length-currBatchSize);
+		
+		//reset
+		timeout = null;
+		logBatch = new Array(batchAllocSize);
+		currBatchSize = 0;
+		
+		//sort the batch by timestamp if necessary - however note that due to the async nature 
+		// of logging you might still send a later batch containing messages with a lower 
+		// logTimestamp. Hence, best avoided if messages will be sorted elsewhere.
+		//toTransmit.sort((a, b) => a.logTimestamp - b.logTimestamp);
+
+		//transmit batch
+		//TODO: Customize batch transmission here
+		//console.log('Batch to transmit: ' + JSON.stringify(toTransmit, null, 2));
+
+		//See SuperAgent docs for request options: https://visionmedia.github.io/superagent/
+		Request
+			.post(LOGGING_ENDPOINT)	
+			.withCredentials()		// set this for CORS requests
+			.type('json')
+			.send({
+				logMessages: toTransmit
+			})
+			.then(res => {
+				console.log("batch tranmitted: ", res);
+			})
+			.catch(err => {
+				console.error("Log batch transmission failure: ", err);
+			});
+	}
+
+	/**
+	 * Create a router listener for log messages. 
+	 * @private
+	 */
+	this.createRouterEndpoints = function () {
+		RouterClient.addListener("logger.service.logMessages", function(error, logMessage) {
+			if (!error) {
+				self.addToBatch(logMessage.data);
+			} else {
+				Logger.error("Failed to setup LoggingExportService listener", error);
+			}
+		});
+	};
+
+	return this;
+};
+
+loggingexportService.prototype = new Finsemble.baseService({
+	startupDependencies: {
+		// Don't depend on any clients or services so that we start-up ASAP
+		services: [],
+		clients: []
```

Comment:
Don't we need to depend on the RouterClient since we use it?

Reply:
No, I believe its created by routerClientInstance so its ready when referenced. Its the first thing up and all other clients depend on it, so depending on any other API client automatically covers it. If you don't need an API you don't need the router client.